### ISSUE TITLE
How can we help?

### ISSUE THREAD

**Issue by andrewrynhard** [ktock/remote-snapshotter#1]
This seems very appealing for usage in https://github.com/talos-systems/talos. I'm curious if there is any way we might be able to help? I would love to help bring this to a state where it is production ready.

**Comment by ktock** [comment]
@andrewrynhard Thanks a lot for your interest!

Any contribution, discussion and bug reporting, etc. are welcome. The discussion is mainly on [#3731@containerd](https://github.com/containerd/containerd/issues/3731).

Could I ask how do you intend to use this remote snapshotter for your products? Currently, we are on the discussion stage, so we keen to hear as many use cases to support as possible.

And I updated the [TODO list on README](https://github.com/ktock/remote-snapshotter#todo). Currently, we are mainly working on containerd's side so we are very happy if you give remote-snapshotter specific issues i.e.:

- Resiliency: Ensure all mounts are available on every Prepare() and report errors when unavailable.
- Auth: Implement auth-related things and the credential management(under the discussion on [#3731@containerd](https://github.com/containerd/containerd/issues/3731))
- Performance: READ performance improvement
- Availability: Especially on NW disconnection

**Comment by ktock** [comment]
@andrewrynhard Are you still interested in this?
Recently this snapshotter supports Kubernetes through CRI.
Can we seek the way for the integration?

**Comment by andrewrynhard** [comment]
@ktock awesome. I saw your blog post. I am interested in how we could. Sounds like we need to get the agent into the OS and then configure containerd to use it. Should be simple enough. Is the agent written in go? Is CGO used?

**Comment by ktock** [comment]
awesome! It's written purly in Go.

**Comment by andrewrynhard** [comment]
Perfect. The primary use case I see is to speed up the pull of the hyperkube image. We will have to create a stargz version of the image?

**Comment by ktock** [comment]
Yes. You can use `ctr-remote` command for converting the original image into stargz and pushing it to registries.

**Comment by andrewrynhard** [comment]
That may be the only issue I can see in getting adoption of this in Talos. Right now we point to the official hyperkube image. But, I think this could still be really neat to have. I will play with it and see how much it improves our boot times.

**Comment by ktock** [comment]
We are seeing an issue for using cloud-provided registries(e.g. GCR). so if you are to try it with these registries, could you wait for #81 get merged? This will be merged today or tomorrow.
DockerHub is well-tested so far.

**Comment by ktock** [comment]
And thanks a lot for trying it out!

**Comment by andrewrynhard** [comment]
Sounds good 👍

**Comment by ktock** [comment]
@andrewrynhard #81 got merged so this should work with cloud-provided registries. Can you try it out? 
You can convert the hyperkube image by `ctr-remote image optimize <HYPERKUBE-IMAGE-PAYH> <YOUR-REPOSITRY-IMAGE-PATH>`.
